src/common_swagger_api/schema/apps/permission.clj

```diff
@@ -0,0 +1,101 @@
+(ns common-swagger-api.schema.apps.permission
+  (:use [common-swagger-api.schema :only [describe ErrorResponse NonBlankString]]
+        [common-swagger-api.schema.apps :only [QualifiedAppId]]
+        [schema.core :only [defschema enum optional-key]]))
+
+(def AppPermissionListingSummary "List App Permissions")
+(def AppPermissionListingDocs
+  "This endpoint allows the caller to list the permissions for one or more apps.
+   The authenticated user must have read permission on every app in the request body for this endpoint to succeed.")
+
+(def AppSharingSummary "Add App Permissions")
+(def AppSharingDocs
+  "This endpoint allows the caller to share multiple apps with multiple users.
+   The authenticated user must have ownership permission to every app in the request body for this endpoint to fully succeed.
+   Note: this is a potentially slow operation and the response is returned synchronously.
+   The DE UI handles this by allowing the user to continue working while the request is being processed.
+   When calling this endpoint, please be sure that the response timeout is long enough.
+   Using a response timeout that is too short will result in an exception on the client side.
+   On the server side, the result of the sharing operation when a connection is lost is undefined.
+   It may be worthwhile to repeat failed or timed out calls to this endpoint.")
+
+(def AppUnsharingSummary "Revoke App Permissions")
+(def AppUnsharingDocs
+  "This endpoint allows the caller to revoke permission to access one or more apps from one or more users.
+   The authenticate user must have ownership permission to every app in the request body for this endoint to fully succeed.
+   Note: like app sharing, this is a potentially slow operation.")
+
+(def AppPermissionEnum (enum "read" "write" "own" ""))
+
+(defschema PermissionListerQueryParams
+  {(optional-key :full-listing)
+   (describe Boolean "If true, include permissions for the authenticated user as well")})
+
+(defschema Subject
+  {:source_id (describe NonBlankString "The identifier of the subject source (for exmaple, 'ldap')")
+   :id        (describe NonBlankString "The subject identifier")})
+
+(defschema AppPermissionListingRequest
```

Comment:
This was named `QualifiedAppIdList` in the apps service, but since nothing else was using it, I renamed it specifically for this endpoint.